### ISSUE TITLE
Auto Leave Inactive Sessions

### ISSUE THREAD

**Issue by kulzae** [Yellow-Dog-Man/Resonite-Issues#1792]
### Is your feature request related to a problem? Please describe.

When world hopping or exploring its possible to be present in quite a few sessions at a time especially if they have long or no afk timeout set. to go though and leave them is extra work that a user might not want to do especially if they don't have an easy way to close them other than going to the world menu and closing each one (if there are multiple hosts for the world you cannot close the one you are in without refocusing and using the close world button)

### Describe the solution you'd like

A setting to leave sessions automatically after x number of minutes, basically a local auto kick. This would only be for sessions you are not hosting to avoid losing work.

### Describe alternatives you've considered

There isnt an alternative as far as i know.

**Comment by Zyro1331** [comment]
I feel that this feature is a decent quick fix for that specific scenario, but another scenario could cause some issues with data loss. An example: You work on your avatar in someones world, someone asks you for help, you pause your work and join the user asking for help. And in the background your client leaves that other world and you end up losing the avatar you were working on due to SimpleAvatarProtection automatically deleting the avatar now that you aren't present. 
What I feel would be more preferred is if the UI by default could better visualize multiple worlds being open like the familiar Web Browser tabs everyone's used to. You wouldn't want your web browser to literally close tabs in the background due to you not visiting them for a little, especially when you don't have an easy way to see and list the places you were previously at. 

I have made a fork of the WorldsYoureInFacet which lays out exactly like the said concept. It by no means is a complete solution to this problem but for your use case it could be very useful. I do wish Resonite could show something like this in the Dash by default, instead of it being completely merged in the Worlds menu.
![image](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/49619936/d77f43a4-6f5a-4f64-bec9-e8b114e57449)
Public Folder Link to said item: `resrec:///U-Zyro1331/R-A3F15A4658B4A94D3CA1161B99A6B51D4712034000828F504EDAF4C5157BE353`

**Comment by paradoxical-autumn** [comment]
I think that while this feature would be helpful as an option, to go along with what @Zyro1331 said, you would need a way to mark worlds you're in as "persistent" so the system wouldn't auto remove you from the world.

This option shouldn't affect the auto afk kick, as that is used for moderation purposes.

**Comment by troyBORG** [comment]
Yeah I love the WorldsYoureInFacet and I feel of something like that was just in the stock dash to show people they still in other worlds if they are new and are unaware when they click on their home tab or something they would see previous things they were in and if they clicked on it all said they would jump back into that world and be like oh I still am in this world and then they could learn about clicking the x on there to close the world once they leave. 

I guess they do have that close world button I think in the top right of the dashboard..  I don't know I removed it so I don't even remember it's there by default...

**Comment by paradoxical-autumn** [comment]
> Yeah I love the WorldsYoureInFacet and I feel of something like that was just in the stock dash to show people they still in other worlds if they are new and are unaware when they click on their home tab or something they would see previous things they were in and if they clicked on it all said they would jump back into that world and be like oh I still am in this world and then they could learn about clicking the x on there to close the world once they leave. 
> 
> I guess they do have that close world button I think in the top right of the dashboard..  I don't know I removed it so I don't even remember it's there by default...

I agree, there should be an official WorldsYourInFacet pinned on the dash by default (and fyi the close world button is there by default)

**Comment by kulzae** [comment]
The issue for the world selector is https://github.com/Yellow-Dog-Man/Resonite-Issues/issues/396

This issue is about an auto leave inactive worlds setting

**Comment by kulzae** [comment]
> > especially if they don't have an easy way to close them other than going to the world menu and closing each one (if there are multiple hosts for the world you cannot close the one you are in without refocusing and using the close world button)
> 
> Can you elaborate on these points, and your other reasons for requesting this setting?
> 
> This issue sounds similar in nature to #538. But if there are underlying issues in how sessions the user are in are presented to the user- those problems should be tackled first rather than applying a setting like this as a bandaid fix to those problems.

There are issues with the world browser that probably contribute here and id need to go make a separate issue for them but i feel this wouldn't be a bandaid fix as much as as it would be a quality of life setting. 

For me personally I prefer to leave sessions if im away for too long and have to manually leave them if the session has no auto kick set.
